pkg/config/config.go

```diff
@@ -316,3 +323,21 @@ func getAllHostnames() ([]string, error) {
 	allHostnames = set.List()
 	return allHostnames, nil
 }
+
+func checkAdvertiseAddressConfigured(advertiseAddress string) error {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return err
+	}
+	for _, addr := range addrs {
+		// interface addresses come with the mask at the end.
+		addrStr := addr.String()
+		if idx := strings.Index(addrStr, "/"); idx != -1 {
+			addrStr = addrStr[:idx]
+		}
+		if addrStr == advertiseAddress {
+			return nil
+		}
```

Comment:
What do you think about using `strings.Contains()`?

Reply:
If I understood you correctly, that could make false matches.
For example, if the advertiseAddress is configured to `10.44.0.1` and one interface has `10.44.0.11/24` then the contains would return true.

Reply:
You're right, thx for clarifying